client/protocols/analytics.py

```diff
@@ -21,6 +25,70 @@ def run(self, messages):
             # TODO(denero) Get the canonical name of the question
             statistics['question'] = self.args.question
 
+        statistics['started'] = self.check_start(messages['file_contents'])
+        print("DEBUG:" + statistics['started'])
+
         messages['analytics'] = statistics
 
+    def check_start(self, files):
+        """returns a dictionary where the key is question number, and the value
+        signals whether the question has been started.
+        """
+        begin = re.compile(r"""
+            [ \t]*
+            \#[ ]BEGIN[ ](.+)    # \1 is the question tag
+            """, re.X | re.M)
+        end = re.compile(r"""
+            [ \t]*
+            \#[ ]END[ ](.+)    # \1 is the question tag
+            """, re.X | re.M)
+        replace = re.compile(r"""
+            \#[ ]Replace[ ].+
+            """, re.X | re.M)
+
+        q_status = {}
+
+        for path in files:
+            lines = files[path].splitlines()
+            if len(lines) == 0:
+                log.warning("File {0} has no content".format(path))
+            line_num = 0
+            started = False
+            in_block = False
+            prev_begin_tag = None
+
+            while line_num < len(lines):
```

Comment:
You can save a few lines by converting this into a `for` loop that uses `enumerate`:

```
for line_num, line in enumerate(lines):
```